server/src/test/java/org/opensearch/index/query/ExistsQueryBuilderTests.java

```diff
@@ -89,9 +89,14 @@ protected ExistsQueryBuilder doCreateTestQueryBuilder() {
         }
         // also sometimes test wildcard patterns
         if (randomBoolean()) {
-            if (randomBoolean()) {
-                fieldPattern = fieldPattern + "*";
-            }
+            fieldPattern = fieldPattern + "*";
+        }
+        // Avoid patterns that would match the derived field "raw.derived_keyword"
+        // This includes "raw", "raw*", "raw.xxx", "r*", "ra*", etc.
+        if (fieldPattern.equals("raw")
```

Comment:
Isn't the equality `(fieldPattern.equals("raw")` case redundant with `fieldPattern.startsWith("raw")`?